public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -370,6 +380,36 @@ function getVariables(oldDash: DashboardSceneState) {
   return variables;
 }
 
+function getAnnotations(state: DashboardSceneState): AnnotationQueryKind[] {
+  const data = state.$data;
+  if (!(data instanceof DashboardDataLayerSet)) {
+    return [];
+  }
+  const annotations: AnnotationQueryKind[] = [];
+  for (const layer of data.state.annotationLayers) {
+    if (!(layer instanceof dataLayers.AnnotationsDataLayer)) {
+      continue;
+    }
+    const result: AnnotationQueryKind = {
+      kind: 'AnnotationQuery',
+      spec: {
+        name: layer.state.query.name,
+        datasource: layer.state.query.datasource!, // FIXME: this cannot be undefined
```

Comment:
If no data source is provided on the query this means it will resolve to the default data source. I think we should loosen the type on the AnnotationQueryKind in order to let the runtime resolve the datasource as it does today.

Alternatively we could assume lack of the data source means "use default data source", but this will be problematic and make the dashboard json less portable. Suddenly the default data source becomes "fixed" in the save model.

Thoughts @ivanortegaalba or @torkelo?

Reply:
Evaluating the two options, I see pros and cons here:

- Use a reserved keyword for default DS (`__default__`) enables differentiation between "no datasource" and "default" since they are two possible scenarios that can exist during runtime. 
- If we export with no DS annotation defined, we should remove that annotation query since this is not reproducible in another system. Also, it is not valid to save. So we could safely assume it exists or remove it. In this case, the default should be already the default ds ref, not a keyword.
- If we import an annotation without DS, we could assume it is the desired behavior by default, but this would be from the provisioning/as-code point of view.

So, in summary, I think `undefined` or `default` should only exist in "import/as-code" scenarios, where we want to fall back to default. Creating a keyword there forces users to learn there is a "keyword" to indicate the system to fall back to default, so I'd consider `undefined` as a possible value to import, but not an existent value to export. So I'd use undefined to indicate in a schema that the DS is the default one.